test/runtime/connectivity.go

```diff
@@ -0,0 +1,273 @@
+package RunT
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/test/helpers"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	log "github.com/sirupsen/logrus"
+)
+
+var _ = Describe("RunConnectivyTest", func() {
+
+	var initilized bool
+	var networkName string = "cilium-net"
+
+	var netperfImage string = "tgraf/netperf"
+	var logger *log.Entry
+	var docker *helpers.Docker
+	var cilium *helpers.Cilium
+
+	initilize := func() {
+		if initilized == true {
+			return
+		}
+		logger = log.WithFields(log.Fields{"test": "RunConnectivyTest"})
+		logger.Info("Starting")
+		docker, cilium = helpers.CreateNewRuntimeHelper("runtime", logger)
+		cilium.WaitUntilReady(100)
+		docker.NetworkCreate(networkName, "")
+		initilized = true
+	}
+
+	BeforeEach(func() {
+		initilize()
+		docker.ContainerCreate("client", netperfImage, networkName, "-l id.client")
+		docker.ContainerCreate("server", netperfImage, networkName, "-l id.server")
+		cilium.Exec("policy delete --all")
+		cilium.EndpointWaitUntilReady()
+		err := helpers.WithTimeout(func() bool {
+			if data, _ := cilium.GetEndpointsNames(); len(data) < 2 {
+				logger.Info("Waiting for endpoints to be ready")
+				return false
+			}
+			return true
+		}, "Endpoints are not ready", &helpers.TimeoutConfig{Timeout: 150})
+		Expect(err).Should(BeNil())
+	}, 150)
+
+	AfterEach(func() {
+		docker.ContainerRm("client")
+		docker.ContainerRm("server")
+		return
+	})
+
+	It("Test containers connectivity without policy", func() {
+		serverData := docker.ContainerInspect("server")
+		serverIP, err := serverData.Filter("{[0].NetworkSettings.Networks.cilium-net.IPAddress}")
+		Expect(err).Should(BeNil())
+		serverIPv6, err := serverData.Filter("{[0].NetworkSettings.Networks.cilium-net.GlobalIPv6Address}")
+		Expect(err).Should(BeNil())
+
+		By("Client can ping to server IPV6")
+		res := docker.ContainerExec("client", fmt.Sprintf("ping6 -c 4 %s", serverIPv6))
+		Expect(res.Correct()).Should(BeTrue())
+
+		By("Client can ping to server Ipv4")
+		res = docker.ContainerExec("client", fmt.Sprintf("ping -c 5 %s", serverIP))
+		Expect(res.Correct()).Should(BeTrue())
+
+		By("Netperf to server from client IPv6")
+		cmd := fmt.Sprintf(
+			"netperf -c -C -t TCP_SENDFILE -H %s", serverIPv6)
+		res = docker.ContainerExec("client", cmd)
+		Expect(res.Correct()).Should(BeTrue())
+	}, 300)
+
+	It("Test containers connectivity WITH policy", func() {
```

Comment:
Perhaps we could share some of the common test code between this and `Test containers connectivity without policy`?